app/models/current_assignment.rb

```diff
@@ -9,23 +10,35 @@ def user_id
     data.fetch(:user_id)
   end
 
-  def assigned_to_user(user)
-    user.id.to_s == user_id
+  def assigned_to_user?(user)
+    assigned? && user.id.to_s == user_id
   end
 
   def assigned?
-    !latest_assignment.nil?
+    !user_id.nil?
   end
 
   private
 
   def data
-    return { user_id: nil, user_name: 'Unassigned' } unless latest_assignment
+    @data ||= projection.run(Rails.application.config.event_store).fetch(:user)
+  end
 
-    latest_assignment.data
+  def projection
+    RailsEventStore::Projection
+      .from_stream("Assigning$#{application.id}")
+      .init(-> { { user: unassigned_user_data } })
+      .when(
+        Assigning::AssignedToUser,
```

Comment:
This method will get pretty long I imagine, would be good to break it up into more discreet methods that are easier to parse: 

```
  def projection
    RailsEventStore::Projection
      .get_stream
      .initialise_with_user
      .when_causes ....

```

Not sure exactly the form to take, but this at the moment looks like a case statement with the logic in the case parts rather than in separate methods.

Something to think about for a future PR in any case 👍

Reply:
Agree